### DISCUSSION TITLE
PHP 8.1 Upgrade

### DISCUSSION THREAD

**Discussion by GaryM0101** [fudforum/FUDforum#21] [General]
Our host provider let us know that PHP 8.1 is available. FUD 3.1.2 was tested under PHP7.4 and 8 but not sure if the sub release 8.1 is compatible. Looks like a lot of performance improvements but some new classes and functions as well.

https://www.php.net/releases/8.1/en.php

We're available to test if needed.

**Comment by naudefj** [comment]
FUDforum is fully compatible with PHP 8.1.
Personally, I would just install the latest code as FUDforum 3.1.2 is already a couple of month old.

Best regards.
Frank

**Comment by GaryM0101** [comment]
We tried upgrading to PHP 8.1 FastCGI. We tried rebuilding all themes for our test and got a depreciation error on the compiler.inc for all the themes except the default.

_Deprecated: Automatic conversion of false to array is deprecated in /home/garmye2/FUDforum/include/compiler.inc on line 653
 Theme grey (en) was successfully rebuilt._

Line 653 is the theme's date format. Then we notice we were getting http 500 errors when posting message replies. The http error logs shows us erroring in the post.php

_[Fri Apr 08 13:18:40.943236 2022] [fcgid:warn] [pid 485943:tid 116437297764096] [client 73.225.60.238:65055] mod_fcgid: stderr: #1 /home/garmye2/vintagegruen.org/vgforum/theme/default/post.php(1209): db_li('INSERT INTO fud...', NULL)_

We decided to revert back to 7.4 FastCGI for now. Note we've been on FUDforum 3.1.2 for a couple of months.
